back-end/config/loader/Firebase.js

```diff
@@ -0,0 +1,47 @@
+const AbstractLoader = require('./AbstractLoader');
+const Firebase = require('../../storage/Firebase');
+
+class Config extends AbstractLoader {
+    async loadConfig() {
+        console.log('[Config] Loading config from Firebase...');
+
+        try {
+            const config = await this.loadConfigFromFirebase();
+
+            this.triggers = config.triggers;
+            this.events = config.events;
+            this.modules = config.modules;
+            this.server = config.server;
+            this.moduleClient = config.moduleClient;
+
+            console.log('[Config] Loaded config.');
+        } catch (error) {
+            console.error('[Config] Unable to load config.');
+            process.exit(1);
+        }
+    }
+
+    async loadConfigFromFirebase() {
+        return await Firebase.load('config').then(function(data) {
+            data = data.toJSON();
+            if (typeof data.triggers === 'undefined') {
```

Comment:
@RickvdStaaij Not sure if I can make this nicer, e.g. more like PHP's array_combine method maybe...

Reply:
you can make an `empty` config, which you then fill with the new object

```js
const emptyConfig = {
    events: [],
    modules: [],
    triggers: [],
    server: [],
    moduleClient: [],
}

return {
    ...emptyConfig,
    ...firebaseConfig,
};
```

Reply:
Fixed it, pretty nicely :+1: